### PR TITLE
support target CLI

### PR DESCRIPTION
**Pull request by d-g-town**
Screenshot 2024-03-28 at 2 18 20 PM
Screenshot 2024-03-28 at 2 18 34 PM

### CODE REVIEW

**Review by stefanmcshane — APPROVED** [review]

File: cli/cmd/commands/target.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 3, 8, 0, '\t', tabwriter.AlignRight)
+
+	fmt.Fprintf(w, "%s\t%s\t%s\n", "ID", "NAME", "CLUSTER ID")
+
+	for _, target := range targets.DeploymentTargets {
+		fmt.Fprintf(w, "%s\t%s\t%d\n", target.ID, target.Name, target.ClusterID)
+	}
+
+	w.Flush()
```
Comment (ianedwards) [inline cli/cmd/commands/target.go:92]: only other thing that might be useful here is whether or not it's the default for a given cluster

**Review by ianedwards — APPROVED** [review]

File: api/client/deployment_target.go [hunk 2]
```diff
@@ -4,28 +4,53 @@ import (
...
+	)
 
-	req := &deployment_target.CreateDeploymentTargetRequest{
-		Selector: selector,
-		Preview:  preview,
-	}
+	return resp, err
+}
+
+// ListDeploymentTargets retrieves all deployment targets in a cluster
```
Comment (ferozemohideen) [inline api/client/deployment_target.go:28]: in a project or cluster?

File: api/server/handlers/deployment_target/create.go [hunk 3]
```diff
@@ -69,14 +56,24 @@ func (c *CreateDeploymentTargetHandler) ServeHTTP(w http.ResponseWriter, r *http
 		return
 	}
 
+	clusterId := request.ClusterId
+	if clusterOk {
```
Comment (ferozemohideen) [inline api/server/handlers/deployment_target/create.go:60]: I feel like if you use the cluster endpoint but also specify cluster id in the request, then request cluster id should take precedence

File: internal/repository/test/deployment_target.go [hunk 4]
```diff
@@ -22,8 +22,13 @@ func (repo *DeploymentTargetRepository) DeploymentTargetBySelectorAndSelectorTyp
 	return nil, errors.New("cannot read database")
 }
 
+// ListForCluster returns all deployment targets for a project
+func (repo *DeploymentTargetRepository) ListForCluster(projectID uint, clusterID uint, preview bool) ([]*models.DeploymentTarget, error) {
```
Comment (ferozemohideen) [inline internal/repository/test/deployment_target.go:26]: would be nice to make cluster id and preview functional arguments

File: cli/cmd/commands/target.go (hunk 1 above)
Comment (d-g-town) [reply cli/cmd/commands/target.go:92]: for sure: will add it in

File: api/server/handlers/deployment_target/create.go (hunk 3 above)
Comment (d-g-town) [reply api/server/handlers/deployment_target/create.go:60]: if there is any auth associated with the cluster scope, I want to stick to that though

File: internal/repository/test/deployment_target.go (hunk 4 above)
Comment (d-g-town) [reply internal/repository/test/deployment_target.go:26]: for sure, will add

Comment (d-g-town) [reply internal/repository/test/deployment_target.go:26]: actually the interface logic and gorm made this annoying: honestly the fix is to move this logic to CCP. Will do later

**Pull request merged**